torchgen/dest/lazy_ir.py

```diff
@@ -307,11 +308,22 @@ def metrics(self, func: NativeFunction, schema: LazyIrSchema) -> str:
 
     def get_device(self, func: NativeFunction, schema: LazyIrSchema) -> str:
         value_args = schema.filtered_args(values=True, scalars=False)
+        scalar_args = schema.filtered_args(values=False, scalars=True)
         value_types_names = [f"{a.name}" for a in value_args if not a.is_wrapped_scalar]
-        assert (
-            len(value_types_names) > 0
-        ), "Code below assumes there is at least one tensor arg"
-        return f"""auto common_device = torch::lazy::GetBackendDevice({', '.join(value_types_names)});
+        if len(value_types_names) > 0:
+            get_device_str = (
+                f"torch::lazy::GetBackendDevice({', '.join(value_types_names)})"
```

Comment:
would an alternative approach be to make GetBackendDevice support a mix of optional and non-optional devices?  then the codegen takes a simple path no matter what?